### PR TITLE
[DPE-3802] make sure backup-restore tests writes on all shards

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Issue
Backup - Restore tests should ensure there are writes to clear / restore on all shards

## Solution
Add writes to all shards

### CODE REVIEW

File: tests/integration/sharding_tests/test_sharding_backups.py [hunk 1]
```diff
@@ -24,19 +23,43 @@
...
+        mongos_client,
+        db_name=SHARD_TWO_DB_NAME,
+        coll_name=SHARD_TWO_COLL_NAME,
+        content={"horse-breed": "unicorn", "real": True},
+    )
+    mongos_client.admin.command("movePrimary", SHARD_TWO_DB_NAME, to=SHARD_TWO_APP_NAME)
+
     yield
     await writes_helpers.clear_db_writes(ops_test)
+    await writes_helpers.remove_db_writes(
```
Comment (Mehdi-Bendriss) [inline tests/integration/sharding_tests/test_sharding_backups.py]: I believe these are exact duplicates, could we extend `clear_db_writes` with default arguments (`db`, `collection`) and make it more generic?

File: tests/integration/sharding_tests/test_sharding_backups.py [hunk 2]
```diff
@@ -232,13 +255,28 @@ async def test_rotate_backup_password(ops_test: OpsTest) -> None:
...
+    shard_two_writes = await writes_helpers.count_shard_writes(
+        ops_test,
+        shard_app_name=SHARD_TWO_APP_NAME,
+        db_name=SHARD_TWO_DB_NAME,
+        collection_name=SHARD_TWO_COLL_NAME,
+    )
+
+    assert cluster_writes["total_writes"], "no writes to backup"
+    assert shard_one_writes, "no writes to backup for shard one"
+    assert shard_two_writes, "no writes to backup for shard two"
```
Comment (Mehdi-Bendriss) [inline tests/integration/sharding_tests/test_sharding_backups.py:274]: Can we add something along the lines of: 
```suggestion
   assert shard_one_writes + shard_two_writes >= cluster_writes["total_writes"], "writes not synced"
```

File: tests/integration/sharding_tests/writes_helpers.py [hunk 3]
```diff
@@ -57,6 +57,26 @@ async def clear_db_writes(ops_test: OpsTest, config_server_name=APP_NAME) -> boo
     client.close()
 
 
+async def remove_db_writes(
```
Comment (Mehdi-Bendriss) [inline tests/integration/sharding_tests/writes_helpers.py:43]: same comment as above about making clear_db_writes generic

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks Mia! I left a few comments

File: tests/integration/sharding_tests/test_sharding_backups.py (hunk 1 above)
Comment (MiaAltieri) [reply tests/integration/sharding_tests/test_sharding_backups.py]: `clear_db_writes` is poorly named - it also stops the continous writes process. 

Let me re-name the functions and consoldate :)

File: tests/integration/sharding_tests/test_sharding_backups.py (hunk 2 above)
Comment (MiaAltieri) [reply tests/integration/sharding_tests/test_sharding_backups.py:274]: yes! changed it to `==` but that was because I changed the count function

**Review by Mehdi-Bendriss — APPROVED** [review]

**Pull request merged**